internal/kube/adaptor/config_init.go

```diff
@@ -2,29 +2,67 @@ package adaptor
 
 import (
 	"context"
+	"fmt"
 	"log"
+	"log/slog"
 	"os"
 	paths "path"
+	"time"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
 
+	"github.com/cenkalti/backoff/v4"
+	internalclient "github.com/skupperproject/skupper/internal/kube/client"
+	"github.com/skupperproject/skupper/internal/kube/secrets"
+	"github.com/skupperproject/skupper/internal/kube/watchers"
 	"github.com/skupperproject/skupper/internal/qdr"
 )
 
-func InitialiseConfig(client kubernetes.Interface, namespace string, path string, routerConfigMap string) error {
+func InitialiseConfig(cli internalclient.Clients, namespace string, path string, routerConfigMap string) error {
 	ctxt := context.Background()
```

Comment:
Should there be a line here that checks to see if the passed in parameter `namespace` is `nil` and if `nil` get it from `cli.GetNamespace()` ?